File: frontend/src/types/supportRequest.ts
export type SupportRequestStatus = 0 | 1 | 2  // Open=0, InProgress=1, Resolved=2

export const SupportRequestStatusLabel: Record<SupportRequestStatus, string> = {
  0: 'Açık',
  1: 'İşlemde',
  2: 'Çözüldü',
}

export const SupportRequestStatusColor: Record<SupportRequestStatus, string> = {
  0: 'bg-yellow-100 text-yellow-800',
  1: 'bg-blue-100 text-blue-800',
  2: 'bg-green-100 text-green-800',
}

export interface SupportRequestSummaryDto {
  id: string
  siteId: string
  userId: string
  unitId: string
  unitDoorNumber?: string
  subject: string
  status: SupportRequestStatus
  createdAt: string
  resolvedAt?: string
}

export interface SupportRequestDetailDto extends SupportRequestSummaryDto {
  description?: string
  updatedAt?: string
}

export interface CreateSupportRequestDto {
  unitId: string
  subject: string
  description?: string
}

export interface UpdateSupportRequestDto {
  subject: string
  description?: string
}

export interface UpdateSupportRequestStatusDto {
  status: SupportRequestStatus
}

File: backend/src/SiteYonetimi.Infrastructure/Entities/Shared/SupportRequest.cs
using SiteYonetimi.Shared.Enums;

namespace SiteYonetimi.Infrastructure.Entities.Shared;

public class SupportRequest
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid SiteId { get; set; }
    public Guid UserId { get; set; }
    public Guid UnitId { get; set; }
    public string Subject { get; set; } = string.Empty;
    public string? Description { get; set; }
    public SupportRequestStatus Status { get; set; } = SupportRequestStatus.Open;
    public DateTime? ResolvedAt { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public Unit Unit { get; set; } = null!;
    public ICollection<SupportRequestComment> Comments { get; set; } = [];
}
